From 3f7dce3a41c07a791c24d58de2a2ccae5ec87228 Mon Sep 17 00:00:00 2001
From: Sivaprasad Tummala <sivaprasad.tummala@amd.com>
Date: Mon, 20 Oct 2025 06:20:11 +0000
Subject: [PATCH 03/11] dpdk: exclude Mellanox CX7 from UIO binding

Add support to skip binding the Mellanox ConnectX-7 (CX7)
network device (vendor ID 0x15b3, device ID 0x1021) to the
UIO driver during DPDK device initialization.

This ensures proper handling of devices that are not
compatible with UIO and may require a mlx5 driver.

Signed-off-by: Sivaprasad Tummala <sivaprasad.tummala@amd.com>
---
 src/plugins/dpdk/device/init.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/plugins/dpdk/device/init.c b/src/plugins/dpdk/device/init.c
index 23a26a415..34e8c2fc6 100644
--- a/src/plugins/dpdk/device/init.c
+++ b/src/plugins/dpdk/device/init.c
@@ -659,6 +659,11 @@ dpdk_bind_devices_to_uio (dpdk_config_main_t * conf)
       {
         continue;
       }
+    /* Mellanox CX7 */
+    else if (d->vendor_id == 0x15b3 && d->device_id == 0x1021)
+      {
+        continue;
+      }
     /* Broadcom NetXtreme S, and E series only */
     else if (d->vendor_id == 0x14e4 &&
 	((d->device_id >= 0x16c0 &&
-- 
2.34.1

